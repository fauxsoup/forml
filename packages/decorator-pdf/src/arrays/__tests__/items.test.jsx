import ArrayComponent from '../items';
import Context from '@forml/context';
import React from 'react';
import ReactPDF from '@react-pdf/renderer';
import { render } from '@testing-library/react';
import * as decorator from '../../';

describe('renders', function () {
    let form;
    let schema;
    let title = 'title';
    let description = 'description';
    let styles;
    let root;
    let header;
    let content;
    let paper;

    beforeEach(function () {
        schema = { type: 'string' };
        form = { type: 'array', items: [{ key: [] }] };
    });

    describe('with form options', function () {
        let fields = {
            disableMargin: [true, false],
        };

        Object.keys(fields).forEach(function (field) {
            fields[field].forEach(function (value) {
                describe(`${field}`, function () {
                    test(`${value}`, function () {
                        form = { ...form, [field]: value };
                        const { container } = render(
                            <ReactPDF.Page>
                                <Context.Provider value={{ decorator }}>
                                    <ArrayComponent
                                        form={form}
                                        title={title}
                                        description={description}
                                    />
                                </Context.Provider>
                            </ReactPDF.Page>
                        );

                        expect(container).toMatchSnapshot();
                    });
                });
            });
        });
    });

    test('with title and description', function () {
        const { container } = render(
            <ReactPDF.Page>
                <Context.Provider value={{ decorator }}>
                    <ArrayComponent
                        form={form}
                        title={title}
                        description={description}
                    />
                </Context.Provider>
            </ReactPDF.Page>
        );

        expect(container).toMatchSnapshot();
    });

    test('with title and no description', function () {
        const { container } = render(
            <ReactPDF.Page>
                <Context.Provider value={{ decorator }}>
                    <ArrayComponent form={form} title={title} />
                </Context.Provider>
            </ReactPDF.Page>
        );

        expect(container).toMatchSnapshot();
    });

    test('with description and no title', function () {
        const { container } = render(
            <ReactPDF.Page>
                <Context.Provider value={{ decorator }}>
                    <ArrayComponent form={form} description={title} />
                </Context.Provider>
            </ReactPDF.Page>
        );

        expect(container).toMatchSnapshot();
    });

    test('with no title or description', function () {
        const { container } = render(
            <ReactPDF.Page>
                <Context.Provider value={{ decorator }}>
                    <ArrayComponent form={form} />
                </Context.Provider>
            </ReactPDF.Page>
        );

        expect(container).toMatchSnapshot();
    });

    test('with relevant formStyles', function () {});
});
