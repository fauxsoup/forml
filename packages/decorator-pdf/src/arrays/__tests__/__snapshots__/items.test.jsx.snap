// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with description and no title 1`] = `
<div>
  <page>
    <view
      style="display: flex; flex: 1; flex-direction: column; border: 1pt solid black; margin: 4pt;"
    >
      <view
        style="display: flex; flex: 1; flex-direction: column;"
      >
        <view
          style="background-color: rgb(238, 238, 238); display: flex; flex-direction: column; padding: 4pt; margin-bottom: 4pt;"
        >
          <text
            style="font-size: 9pt; font-weight: 200; margin-top: 2pt;"
          >
            title
          </text>
        </view>
        <view />
      </view>
    </view>
  </page>
</div>
`;

exports[`renders with form options disableMargin false 1`] = `
<div>
  <page>
    <view
      style="display: flex; flex: 1; flex-direction: column; border: 1pt solid black; margin: 4pt;"
    >
      <view
        style="display: flex; flex: 1; flex-direction: column;"
      >
        <view
          style="background-color: rgb(238, 238, 238); display: flex; flex-direction: column; padding: 4pt; margin-bottom: 4pt;"
        >
          <text
            style="font-size: 12pt; font-weight: 400;"
          >
            title
          </text>
          <text
            style="font-size: 9pt; font-weight: 200; margin-top: 2pt;"
          >
            description
          </text>
        </view>
        <view />
      </view>
    </view>
  </page>
</div>
`;

exports[`renders with form options disableMargin true 1`] = `
<div>
  <page>
    <view
      style="display: flex; flex: 1; flex-direction: column; border: 1pt solid black; margin: 0px;"
    >
      <view
        style="display: flex; flex: 1; flex-direction: column;"
      >
        <view
          style="background-color: rgb(238, 238, 238); display: flex; flex-direction: column; padding: 4pt; margin-bottom: 4pt;"
        >
          <text
            style="font-size: 12pt; font-weight: 400;"
          >
            title
          </text>
          <text
            style="font-size: 9pt; font-weight: 200; margin-top: 2pt;"
          >
            description
          </text>
        </view>
        <view />
      </view>
    </view>
  </page>
</div>
`;

exports[`renders with no title or description 1`] = `
<div>
  <page>
    <view
      style="display: flex; flex: 1; flex-direction: column; border: 1pt solid black; margin: 4pt;"
    >
      <view
        style="display: flex; flex: 1; flex-direction: column;"
      >
        <view />
      </view>
    </view>
  </page>
</div>
`;

exports[`renders with title and description 1`] = `
<div>
  <page>
    <view
      style="display: flex; flex: 1; flex-direction: column; border: 1pt solid black; margin: 4pt;"
    >
      <view
        style="display: flex; flex: 1; flex-direction: column;"
      >
        <view
          style="background-color: rgb(238, 238, 238); display: flex; flex-direction: column; padding: 4pt; margin-bottom: 4pt;"
        >
          <text
            style="font-size: 12pt; font-weight: 400;"
          >
            title
          </text>
          <text
            style="font-size: 9pt; font-weight: 200; margin-top: 2pt;"
          >
            description
          </text>
        </view>
        <view />
      </view>
    </view>
  </page>
</div>
`;

exports[`renders with title and no description 1`] = `
<div>
  <page>
    <view
      style="display: flex; flex: 1; flex-direction: column; border: 1pt solid black; margin: 4pt;"
    >
      <view
        style="display: flex; flex: 1; flex-direction: column;"
      >
        <view
          style="background-color: rgb(238, 238, 238); display: flex; flex-direction: column; padding: 4pt; margin-bottom: 4pt;"
        >
          <text
            style="font-size: 12pt; font-weight: 400;"
          >
            title
          </text>
        </view>
        <view />
      </view>
    </view>
  </page>
</div>
`;
